### PR TITLE
add ident.ident lookup in refinement expressions in paxel mode

### PR DESCRIPTION
**Pull request by cromwellian**

### CODE REVIEW

File: src/runtime/manifest-parser.pegjs [hunk 1]
```diff
@@ -678,9 +698,16 @@ NameWithColon
...
+  = '=' {
+    pushPaxelMode(true);
+    return text();
+  }
+
 ParticleHandleConnectionBody
-  = name:NameWithColon? direction:(Direction '?'?)? whiteSpace type:ParticleHandleConnectionType annotations:SpaceAnnotationRefList? maybeTags:SpaceTagList? expression:(whiteSpace? '=' multiLineSpace PaxelExpression)?
+  = name:NameWithColon? direction:(Direction '?'?)? whiteSpace type:ParticleHandleConnectionType annotations:SpaceAnnotationRefList? maybeTags:SpaceTagList? expression:(whiteSpace? PaxelMode multiLineSpace PaxelExpression)?
   {
+    popPaxelMode();
```
Comment (piotrswigon) [inline src/runtime/manifest-parser.pegjs:710]: should you only pop() if expression was actually specified? What if expression was not defined for the connection?

File: src/runtime/manifest-parser.pegjs [hunk 2]
```diff
@@ -1833,9 +1854,14 @@ PrimaryExpression
...
   }
-  / fn: fieldName
+  / fn: fieldName nested:('.' fieldName)*
   {
-    return toAstNode<AstNode.FieldNode>({kind: 'field-name-node', value: fn});
+    // nested is ignored, used only to allow Paxel expressions to parse as text
+    if (!isPaxelMode() && nested && nested.length > 0) {
+      error('Scope lookups are not permitted for refinements, only in paxel expressions.');
+    } else {
+      return toAstNode<AstNode.FieldNode>({kind: 'field-name-node', value: fn});
```
Comment (piotrswigon) [inline src/runtime/manifest-parser.pegjs:1863]: where is `nested` appearing here?

Also, I think the AST in Kotlin for a.b.c is:
```
FieldExpr(
  var: "c",
  qualifier =  FieldExpr(
    var: "b",
    FieldExpr(
      var: "a",
      qualifier: null
    )
  )
)
```

You probably **don't want** it the other way around here, i.e.
```
FieldExpr(
  var: "a",
  qualifier =  FieldExpr(
    var: "b",
    FieldExpr(
      var: "c",
      qualifier: null
    )
  )
)
```

File: src/runtime/tests/manifest-parser-test.ts [hunk 3]
```diff
@@ -854,6 +861,14 @@ describe('manifest parser', () => {
         baz: reads Baz {z: Number}
       `);
     });
+    it('parses from/select expression with new and scope lookup', () => {
```
Comment (piotrswigon) [inline src/runtime/tests/manifest-parser-test.ts:864]: Maybe also:
```
it('allows expressions in new entity selection', () => {
      parse(`
      particle Converter
        foo: reads Foo {x: Number}
        bar: writes Bar {y: Number} = from p in foo.x select new Bar {y: p.num / 2}
      `);
    });
```

**Review by piotrswigon — APPROVED** [review]

File: src/runtime/manifest-parser.pegjs (hunk 2 above)
Comment (piotrswigon) [reply src/runtime/manifest-parser.pegjs:1863]: Oh, sorry, I've read the comment... we don't care about AST structure here yet. Ignore me.

File: src/runtime/manifest-parser.pegjs (hunk 1 above)
Comment (cromwellian) [reply src/runtime/manifest-parser.pegjs:710]: good catch, yes.

**Pull request merged**